src/gui/src/gui.cpp

```diff
@@ -476,6 +484,28 @@ int Gui::select(const std::string& type,
   logger_->error(utl::GUI, 35, "Unable to find descriptor for: {}", type);
 }
 
+void Gui::netAttributeFilter(const std::string& attribute,
```

Comment:
I was envisioning this to be more generic.  The descriptor has getProperties that lets you traverse them generically.  If the property name matches the 'attribute' then compare the property value to 'value'.  I think you could cover all types/properties in a carefully constructed method rather than having to enumerate every case.